### PR TITLE
Use some if constexpr to reduce the code

### PR DESCRIPTION
**Pull request by aprokop**
@dalg24 wanted to run this past you before creating a public PR.

### CODE REVIEW

File: src/details/ArborX_DetailsMortonCode.hpp [hunk 1]
```diff
@@ -192,47 +192,35 @@ KOKKOS_INLINE_FUNCTION unsigned int morton32(Point const &p)
...
-
   using KokkosExt::max;
   using KokkosExt::min;
 
-  unsigned long long r = 0;
-  for (int d = 0; d < DIM; ++d)
+  constexpr int DIM = GeometryTraits::dimension_v<Point>;
+  constexpr unsigned N = (1u << (63 / DIM));
+
+  if constexpr (DIM > 2)
```
Comment (dalg24) [inline src/details/ArborX_DetailsMortonCode.hpp:208]: Original code was doing `DIM != 2`, please confirm you meant to change this and make sure the upcoming PR description mentions it if the answer is yes.

File: src/details/ArborX_DetailsMortonCode.hpp [hunk 2]
```diff
@@ -192,47 +192,35 @@ KOKKOS_INLINE_FUNCTION unsigned int morton32(Point const &p)
...
+      r += (expandBitsBy<DIM - 1>((unsigned long long)x) << (DIM - d - 1));
+    }
+    return r;
   }
+  else
+  {
+    // Special case because it needs double. Float is not sufficient to
+    // represent large integers, which would result in some missing bins.
+    auto xd = min(max((double)p[0] * N, 0.), (double)N - 1);
+    auto yd = min(max((double)p[1] * N, 0.), (double)N - 1);
```
Comment (dalg24) [inline src/details/ArborX_DetailsMortonCode.hpp:223]: Would def not be right for `DIM==1`

File: src/geometry/ArborX_HyperBox.hpp [hunk 3]
```diff
@@ -68,32 +68,28 @@ struct Box
...
+        maxCorner()[d] = max(maxCorner()[d], other.maxCorner()[d]);
+      }
+      else if constexpr (GeometryTraits::is_point<Geometry>{})
+      {
+        minCorner()[d] = min(minCorner()[d], other[d]);
+        maxCorner()[d] = max(maxCorner()[d], other[d]);
+      }
+      else
+      {
+        assert(false);
```
Comment (dalg24) [inline src/geometry/ArborX_HyperBox.hpp]:
```suggestion
        static_assert(std::is_void_v<Geometry>, "current implementation only supporting point or box");
```

File: src/details/ArborX_DetailsUtils.hpp [hunk 4]
```diff
@@ -74,18 +74,44 @@ create_layout_right_mirror_view(
...
-          (View::rank == 1 && !std::is_same<typename View::traits::array_layout,
-                                            Kokkos::LayoutStride>::value)) &&
-         std::is_same<typename View::traits::memory_space,
-                      typename ExecutionSpace::memory_space>::value)>::type * =
-        nullptr)
-{
-  return src;
+inline auto
+create_layout_right_mirror_view(ExecutionSpace const & /*execution_space*/,
+                                View const &src)
```
Comment (dalg24) [inline src/details/ArborX_DetailsUtils.hpp]: This is fine but it reminds me how this function could use some refactor/cleanup

File: src/details/ArborX_SpaceFillingCurves.hpp [hunk 5]
```diff
@@ -27,43 +27,37 @@ namespace Experimental
...
   KOKKOS_FUNCTION auto operator()(Box const &scene_bounding_box,
                                   Geometry const &geometry) const
   {
-    auto p = Details::returnCentroid(geometry);
+    auto p = [&geometry]() {
+      if constexpr (GeometryTraits::is_point<Geometry>{})
+        return geometry;
+      else
+        return Details::returnCentroid(geometry);
+    }();
```
Comment (dalg24) [inline src/details/ArborX_SpaceFillingCurves.hpp]: I really dislike this.  Do we really need the code specialization?

File: src/details/ArborX_DetailsMortonCode.hpp (hunk 1 above)
Comment (aprokop) [reply src/details/ArborX_DetailsMortonCode.hpp:208]: I added static asserts to make sure `DIM >= 2`. We don't support (and don't think we should) `DIM = 1` case.

**Pull request closed without merging**